### ISSUE TITLE
Advisory for v7

### ISSUE THREAD

**Issue by tombrown-ibm** [postcss/postcss#1574]
Would it be possible or is there a plan to patch v7 for the following advisory?

https://www.npmjs.com/advisories/1693

**Comment by tombrowndev** [comment]
Here's my attempt

https://github.com/postcss/postcss/pull/1575

**Comment by ai** [comment]
Nope. I supported PostCSS 7 for a 6 month after PostCSS 8 support. The support was ended at January 1, 2021.

Only commercial support is possible right now.

**Comment by tombrowndev** [comment]
Ok, feel free to close

**Comment by stephane-arista** [comment]
react-scripts relies on v7, so this is probably breaking a ton of peoples environments, and there is no straightforward upgrade path.

**Comment by Hypnosphi** [comment]
Here's the PR for reference https://github.com/facebook/create-react-app/pull/9716

I hoped it would include some replacement for `postcss-preset-env` though, as it's still depends on v7

**Comment by stephane-arista** [comment]
So what's the problem with just patching v7, since it's a very minor fix. (from my quick look)

**Comment by ai** [comment]
> So what's the problem with just patching v7, since it's a very minor fix. (from my quick look)

Releasing and then supporting the issue.

In open source you need about 3-5x more work in additional to writing a code. Everything can go bad, and then I will fix instead of the sleep (for instance, I didn’t sleep well today because I fixed bug after `caniuse-lite` security fixes).

**Comment by Hypnosphi** [comment]
You have a section in README that looks like a list of recommended plugins, and it has quite a number of packages depending on the unsupported v7. stylelint and postcss-preset-env are some of the notable examples. Should they be removed from the list or marked somehow?

https://github.com/postcss/postcss#plugins

**Comment by ai** [comment]
> Should they be removed from the list or marked somehow?

The better way is helping them in migration by sending PRs. But if you will find a dead plugin, which will ignore PRs, I will remove it from the list.

**Comment by dezfowler** [comment]
@ai I'm not asking you to work more but we do need to find a solution. Unfortunately a successful open source project can be both a blessing and a curse for a maintainer.

Those ~10 other libraries migrating to v8 is not a simple ask - this issue coming up will no doubt accelerate that migration but it's a process that's going to take a while when there are chains of dependencies involved. 

I appreciate that you personally no longer want to invest time supporting v7 however are there other options? Are you the sole person with commit/publish rights or are there other core contributors?

**Comment by athix** [comment]
This issue is also affecting @rails/webpacker: https://github.com/rails/webpacker/issues/3017

**Comment by web-padawan** [comment]
Stylelint is still in progress of migration to v8 https://github.com/stylelint/stylelint/issues/4942 and they mentioned it won't be easy.
Looks like we have to ignore those dependabot alerts until new Stylelint major version is out 🤷‍♂️ 

This seems a bit weird as the suggested fix is to cherry-pick 2 lines of code but it can't be done per above comment.

**Comment by ai** [comment]
I backported ReDoS fix from 8.x to 7.0.36 and ask to change CVE (it could take a few days).

I fixed it only because other big open source project which I respect did it. I will not do it again.

Migrate from PostCSS 7 and donate to awesome (but very underfunded) Stylelint and cssnano to help them in PostCSS 8 migration. Our entire industry is based on their work:
* https://opencollective.com/stylelint
* https://opencollective.com/cssnano
